connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/TestPlugins.java

```diff
@@ -98,22 +97,56 @@
      * to load internal classes, and samples information about their initialization.
      */
     public static final String SERVICE_LOADER = "test.plugins.ServiceLoaderPlugin";
+    /**
+     * Class name of a plugin which reads a version string from resource.
+     */
+    public static final String READ_VERSION_FROM_RESOURCE = "test.plugins.ReadVersionFromResource";
 
     private static final Logger log = LoggerFactory.getLogger(TestPlugins.class);
-    private static final Map<String, File> PLUGIN_JARS;
+    private static final Map<String, PluginJar> PLUGIN_JARS;
     private static final Throwable INITIALIZATION_EXCEPTION;
 
+    private static final class PluginJar {
+        String className;
+        File jarFile;
+
+        public PluginJar(String className, File jarFile) {
```

Comment:
Perhaps another thing that should be representable by the data structure in this class is a jar file which contains multiple plugins. This is a completely supported use-case, but we don't have any tests related to it. If we're adding the ability to represent an unsupported configuration, maybe we should make sure it's able to represent other supported use-cases too :)